test-cypress/integration/sparql/sparql.menu.spec.js

```diff
@@ -1,807 +1,1034 @@
+        return cy.get('#yasr .nav.pagination');
+    }
+
+    function getResultPages() {
+        return getResultPagination().find('li')
+    }
+
+    function getModal() {
+        return cy.get('.modal');
+    }
+
+    const TABS_SELECTOR = '#sparql-content .nav-tabs .sparql-tab';
+
+    function getTabs() {
+        return cy.get(TABS_SELECTOR);
+    }
+
+    function getActiveTab() {
+        return cy.get(TABS_SELECTOR + '.active')
+    }
+
+    function getActiveTabLink() {
+        return cy.get(TABS_SELECTOR + '.active .nav-link')
+    }
+
+    function getNewTabButton() {
+        return cy.get('#wb-sparql-addNewTab');
+    }
+
+    function getLastTab() {
+        return getTabs().last();
+    }
+
+    function renameTab(position, newName) {
+        getTabs().eq(position - 1).then(tab => {
+            cy.wrap(tab)
+            // First click is to focus it in case it's not the active tab
+                .click()
+                .find('.nav-link')
+                .dblclick();
+            cy.wrap(tab)
+                .find('.editable-input')
+                .type(newName)
+                .should('have.value', newName);
+            cy.wrap(tab)
+                .find('.editable-buttons .btn-primary')
+                .click()
+                .should('not.exist');
+        });
+    }
+
+    function verifyTabName(position, name) {
+        getTabs().eq(position - 1)
+            .should('be.visible')
+            .find('.nav-link')
+            .should('have.text', name);
+    }
+
+    function getTabCloseBtn(position) {
+        return getTabs().eq(position - 1).find('.delete-sparql-tab-btn');
+    }
+
+    function getModalConfirmBtn() {
+        return getModal().find('.modal-footer .btn-primary');
+    }
+
+    function verifyQueryAreaContains(query) {
+        // Using the CodeMirror instance because getting the value from the DOM is very cumbersome
+        getQueryArea().then(codeMirrorEl => {
+            const cm = codeMirrorEl[0].CodeMirror;
+            expect(cm.getValue().includes(query)).to.be.true;
+        });
+    }
 
-        // Execute the same query
-        cy.get('div#runButton > #wb-sparql-runQuery').click();
+    function getQueryArea() {
+        return cy.get('#queryEditor .CodeMirror');
+    }
 
+    function getQueryTextArea() {
+        return getQueryArea().find('textarea');
+    }
 
-        // Verify that there are two results displaying the links between ":a owl:sameAs :a" and ":a owl:sameAs :b"
-        cy.verifyQueryResultsS(2);
+    function getNoQueryRun() {
+        return cy.get('#yasr-inner .no-query-run');
+    }
+
+    function clearQuery() {
+        // Using force because the textarea is not visible
+        getQueryTextArea().type('{ctrl}a{backspace}', {force: true});
+    }
+
+    function typeQuery(query) {
+        clearQuery();
+        getQueryTextArea().type(query, {force: true, parseSpecialCharSequences: false});
+    }
+
+    function pasteQuery(query) {
```

Comment:
Well it appeared we have a paste functionality already. Why we don't use it more extensively?

Reply:
I tended using it for bigger queries.

Reply:
If we add exclusive test that verifies correct typing, we could replace most of it with the paste method (excluding the automatic prefix insertion which doesn't work with pasting :) )